scripts/performance-tests.sh

```diff
@@ -0,0 +1,150 @@
+  echo "Using secret defined in ${GOOGLE_APPLICATION_CREDENTIALS}"
+  gcloud auth activate-service-account ${user_name} --key-file=${GOOGLE_APPLICATION_CREDENTIALS}
+}
+
+# Update resources installed on the cluster.
+# Parameters: $1 - cluster name
+#             $2 - cluster zone/region
+function update_cluster() {
+  gcloud container clusters get-credentials $1 --zone=$2 --project=${PROJECT_NAME} || abort "failed to get cluster creds"
+  # Set up the configmap to run benchmarks in production
+  echo ">> Setting up 'prod' config-mako on cluster $1 in zone $2"
+  cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-mako
+data:
+  # This should only be used by our performance automation.
+  environment: prod
+EOF
+  # Create secrets required for running benchmarks on the cluster
+  echo ">> Creating secrets on cluster $1 in zone $2"
+  kubectl create secret generic mako-secrets \
+    --from-file=robot.json=${GOOGLE_APPLICATION_CREDENTIALS} \
+    --from-file=github-token=${GITHUB_TOKEN} \
+    --from-file=slack-read-token=${SLACK_READ_TOKEN} \
+    --from-file=slack-write-token=${SLACK_WRITE_TOKEN}
+  # Delete all benchmark jobs to avoid noise in the update process
+  echo ">> Deleting all cronjobs and jobs on cluster $1 in zone $2"
+  kubectl delete cronjob --all
+  kubectl delete job --all
+  
+  if function_exists update_knative; then
+    update_knative || abort "failed to update knative"
+  fi
+  # get benchmark name from the cluster name
+  local benchmark_name=$(get_benchmark_name $1)
+  if function_exists update_benchmark; then
+    update_benchmark ${benchmark_name} || abort "failed to update benchmark"
+  fi
+}
+
+# Get benchmark name from the cluster name.
+# Parameters: $1 - cluster name
+function get_benchmark_name() {
+  # get benchmark_name by removing the prefix from cluster name, e.g. get "load-test" from "serving-load-test"
+  echo ${1#$REPO_NAME"-"}
+}
+
+# Update the clusters related to the current repo.
+function update_clusters() {
+  header "Updating all clusters for ${REPO_NAME}"
+  local all_clusters=$(gcloud container clusters list --project="${PROJECT_NAME}" --format="csv[no-heading](name,zone)")
+  echo ">> Project contains clusters:" ${all_clusters}
+  for cluster in ${all_clusters}; do
+    local name=$(echo "${cluster}" | cut -f1 -d",")
+    # the cluster name is prefixed with repo name, here we should only handle clusters related to the current repo
+    [[ ! ${name} =~ ^${REPO_NAME} ]] && continue
+    local zone=$(echo "${cluster}" | cut -f2 -d",")
+
+    # Update all resources installed on the cluster
+    update_cluster ${name} ${zone}
+  done
+  header "Done updating all clusters"
+}
+
+# Delete the old clusters related to the current repo, and recreate them with the same configuration.
+function recreate_clusters() {
+  header "Recreating clusters for ${REPO_NAME}"
+  go run $GOPATH/src/knative.dev/test-infra/pkg/perf-cluster-operation \
```

Comment:
This is an absolute location, meaning that any repo sourcing this script must have test-infra checked out in order to run this command. I believe you intend to have the tool vendored instead. Here and everywhere.

Reply:
Changed to `go run knative.dev/test-infra/tools/perf-ops`